From: Cyrill Gorcunov <gorcunov@openvz.org>
Date: Wed, 8 Dec 2010 14:12:39 +0000 (+0300)
Subject: net, iptables: Restore setting VE_IP_NAT bit in running modules mask
X-Git-Url: http://git.openvz.org/?p=linux-2.6.32-openvz;a=commitdiff_plain;h=04e9b2f949ad9c2ded425caecc17341d5300f2b5

net, iptables: Restore setting VE_IP_NAT bit in running modules mask

During migration to new iptables management code the setting of
VE_IP_NAT bit was lost. In the former code it set at module symbol
resolving time (which was cut off eventually). Restore this
functionality by setting this bit if VE is allowed to.

http://bugzilla.openvz.org/show_bug.cgi?id=1603

Signed-off-by: Cyrill Gorcunov <gorcunov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/net/ipv4/netfilter/nf_nat_core.c b/net/ipv4/netfilter/nf_nat_core.c
index 2c77ffe..96f07a4 100644
--- a/net/ipv4/netfilter/nf_nat_core.c
+++ b/net/ipv4/netfilter/nf_nat_core.c
@@ -682,6 +682,9 @@ nfnetlink_parse_nat_setup(struct nf_conn *ct,
 
 static int __net_init nf_nat_net_init(struct net *net)
 {
+	if (net_ipt_permitted(net, VE_IP_NAT))
+		net_ipt_module_set(net, VE_IP_NAT);
+
 	/* Leave them the same for the moment. */
 	net->ipv4.nat_htable_size = net->ct.htable_size;
 	net->ipv4.nat_bysource = nf_ct_alloc_hashtable(&net->ipv4.nat_htable_size,
